### PR TITLE
[SYCL] Add number HW threads per EU query

### PR DESCRIPTION
**Pull request by MrSidims**
The appropriate spec update: https://github.com/intel/llvm/pull/4876
Test: https://github.com/intel/llvm-test-suite/pull/550

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

File: sycl/plugins/cuda/pi_cuda.cpp
Comment (steffenlarsen) [inline sycl/plugins/cuda/pi_cuda.cpp:1678]: For consistency, would you mind adding this case to pi_hip.cpp and pi_opencl.cpp as well?

**Review by steffenlarsen — DISMISSED** [review]
Just a minor request, but otherwise LGTM.

File: sycl/plugins/cuda/pi_cuda.cpp
Comment (MrSidims) [reply sycl/plugins/cuda/pi_cuda.cpp:1678]: Sure, thanks!

**Comment by MrSidims** [comment]
Tentatively converted to draft since I want https://github.com/intel/llvm/pull/4876 be approved first before merging the implementation.

File: sycl/plugins/opencl/pi_opencl.cpp
Comment (MrSidims) [inline sycl/plugins/opencl/pi_opencl.cpp:187]: There is actually an extension, that adds these queries for OpenCL: https://github.com/KhronosGroup/OpenCL-Docs/blob/master/extensions/cl_intel_device_attribute_query.asciidoc
I plan to add these queries to pi_opencl all together a bit later.

**Review by steffenlarsen — DISMISSED** [review]

**Comment by MrSidims** [comment]
@smaslov-intel @againull could you please take a look?

File: sycl/plugins/opencl/pi_opencl.cpp
Comment (MrSidims) [reply sycl/plugins/opencl/pi_opencl.cpp:187]: 'A bit later' is prolonged since even if the extension is public - it's yet not in OpenCL register and the headers and hence my local build for this implementation is failing.

**Comment by MrSidims** [comment]
@smaslov-intel ping

File: sycl/plugins/opencl/pi_opencl.cpp
Comment (smaslov-intel) [reply sycl/plugins/opencl/pi_opencl.cpp:187]: Maybe add a TODO comment naming the plans

File: sycl/include/CL/sycl/detail/pi.h
Comment (smaslov-intel) [inline sycl/include/CL/sycl/detail/pi.h:304]: why did you leave the gap? why not 0x10112?

Comment (MrSidims) [reply sycl/include/CL/sycl/detail/pi.h:304]: Hm, I was under impression, that this value is claimed by one of other CL_ definition, but it does look like I was wrong or misinterpreted compilation error.

File: sycl/plugins/opencl/pi_opencl.cpp
Comment (MrSidims) [reply sycl/plugins/opencl/pi_opencl.cpp:189]: Done

**Review by smaslov-intel — APPROVED** [review]

**Comment by MrSidims** [comment]
Fixed ABI test. @bader guess I have both required approvals.

**Review by smaslov-intel — APPROVED** [review]

**Comment by bader** [comment]
> Fixed ABI test. @bader , guess I have both required approvals.

The https://github.com/intel/llvm-test-suite/pull/550 validation status is unclear. I see a lot of failures. Are they caused by this patch?

**Review by steffenlarsen — APPROVED** [review]

**Comment by MrSidims** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/606

**Comment by MrSidims** [comment]
@bader @vladimirlaz are we good to merge the PRs ? Assertions seems to be a known issue. I'll report mismatches between 505 and 606 test PR results internally.

**Pull request merged**
